error_formatter.go

```diff
@@ -0,0 +1,91 @@
+/*
+Copyright 2017 Google Inc. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package jsonnet
+
+import (
+	"bytes"
+	"fmt"
+)
+
+type ErrorFormatter struct {
+	// TODO(sbarzowski) use this
+	// MaxStackTraceSize  is the maximum length of stack trace before cropping
+	MaxStackTraceSize int
+	// TODO(sbarzowski) use these
+	pretty   bool
+	colorful bool
+	SP       SourceProvider
```

Comment:
a) For now I think we won't expose it in libjsonnet api. We can add it in the future. For now it's just very convenient for me now and long-term I think it's also useful for users so I want it to start capitalizing early.
b) I already have a branch which starts to use this stuff.